src/tribler-core/tribler_core/components/tag/tag_component.py

```diff
@@ -0,0 +1,43 @@
+from ipv8.peerdiscovery.discovery import RandomWalk
+from ipv8_service import IPv8
+
+from tribler_core.components.base import Component
+from tribler_core.components.gigachannel.community.sync_strategy import RemovePeers
+from tribler_core.components.ipv8 import Ipv8Component
+from tribler_core.components.metadata_store.metadata_store_component import MetadataStoreComponent
+from tribler_core.components.tag.community.tag_community import TagCommunity
+
+INFINITE = -1
+
+
+class TagComponent(Component):
+    community: TagCommunity
+    _ipv8: IPv8
+
+    async def run(self):
+        await super().run()
+
+        ipv8_component = await self.require_component(Ipv8Component)
+        metadata_store_component = await self.require_component(MetadataStoreComponent)
+
+        self._ipv8 = ipv8_component.ipv8
+
+        community = TagCommunity(
+            ipv8_component.peer,
+            self._ipv8.endpoint,
+            self._ipv8.network,
+            metadata_store=metadata_store_component.mds
+        )
+        self.community = community
+
+        self._ipv8.add_strategy(community, RandomWalk(community), 30)
+        self._ipv8.add_strategy(community, RemovePeers(community), INFINITE)
```

Comment:
**issue:** The target peers of the `RandomWalk` are equal to the default `Community` `max_peers`. Therefore, this code should also follow the default target peers of `20` instead of `30`.

**issue:** This code is removing peers from the common peer pool `self._ipv8.network`. There should really only ever be one strategy removing from the common peer pool (which is already handled by the `DiscoveryCommunity` churn strategy and modulated by the `GoldenRatio` strategy). This line should be removed.

**suggestion:**
```suggestion
        self._ipv8.add_strategy(community, RandomWalk(community), 20)
```

**thought:** I've seen these issues before in other PRs. Perhaps the default way to add new `Community` instances should be in some common base class or perhaps as a method in the IPv8 component?

Reply:
Thank you for your explanation. The code has been fixed.

You are right it is not clear, what is the best way to create just a default `Community`

>should be in some common base class or perhaps as a method in the IPv8 component?

Something like this?

```python
class Ipv8Component(RestfulComponent):
    ipv8: IPv8
    peer: Peer
    dht_discovery_community: Optional[DHTDiscoveryCommunity] = None

    ...

    def add_default_community(self, community):
        self._ipv8.add_strategy(community, RandomWalk(community), 20)
        community.bootstrappers.append(self.make_bootstrapper())
```

Reply:
Yes, that should work.

One nuance is that if---at any point in the future---someone makes any part of this code multithreaded, the bootstrapper should be added to the `community` before calling `add_strategy`. Switching the two lines in `add_default_community` would be "safer".

Reply:
OK, cool! I can apply it to all components.

Reply:
Applied here: https://github.com/Tribler/tribler/pull/6402

All the changes have been applied to this branch as well.